docs/content/docs/security/ssl.md

```diff
@@ -0,0 +1,367 @@
+application-specific external services such as sources and sinks may use some other network protocol.
+
+The server will, by default, accept connections from any client, meaning that the REST endpoint does 
+not authenticate the client. These REST endpoints, however, can be configured to require SSL encryption 
+and mutual authentication. 
+
+However, the recommended approach is setting up and configuring a dedicated proxy service (a "sidecar 
+proxy") that controls access to the REST endpoint. This involves binding the REST endpoint to the 
+loopback interface (or the pod-local interface in Kubernetes) and starting a REST proxy that authenticates 
+and forwards the requests to Flink. Examples for proxies that Flink users have deployed are [Envoy Proxy](https://www.envoyproxy.io/) 
+or [NGINX with MOD_AUTH](http://nginx.org/en/docs/http/ngx_http_auth_request_module.html).
+
+The rationale behind delegating authentication to a proxy is that such proxies offer a wide variety
+of authentication options and thus better integration into existing infrastructures.
+
+## Queryable State
+
+Connections to the [queryable state]({{< ref "docs/dev/datastream/fault-tolerance/queryable_state" >}}) 
+endpoints is currently not authenticated or encrypted.
+
+## SSL Setups
+
+{{< img src="/fig/ssl_mutual_auth.svg" alt="SSL Mutual Authentication" width=75% >}}
+
+Each participant has a keystore and a truststore, which are files. 
+
+A keystore contains a certificate (which contains a public key) and a private key. A truststore 
+contains trusted certificates and certificate chains/authorities. 
+
+Establishing encrypted, authenticated communication is a multi-step process, shown in the figure. 
+Certificates are exchanged and validated against the truststore, after which the two parties can 
+safely communicate.
+
+### Typical SSL Setup in Flink
+
+For mutually authenticated internal connections, note that:
+
+- a keystore and a truststore can contain the same dedicated certificate 
+- the same file can be used for both keystore and truststore
+- wildcard hostnames or addresses can be used 
+
+For internal communication between servers in a Flink cluster, a secure setup can be established with 
+a single, self-signed certificate that all parties use as both their keystore and truststore. You can 
+also use this approach for external communication when establishing mutual authentication for communication 
+between clients and the Flink Master.
+
+### Configuring Keystores and Truststores
+
+The SSL configuration requires configuring a keystore and a truststore such that the truststore trusts
+the keystore's certificate.
+
+You can use the [keytool utility](https://docs.oracle.com/javase/8/docs/technotes/tools/unix/keytool.html) 
+to generate keys, certificates, keystores, and truststores:
+
+```bash
+    keytool -genkeypair -alias flink.internal -keystore internal.keystore \
```

Comment:
Is there a specific reason why this keytool command is broken with `\` and all the others not?